assets/javascripts/discourse/initializers/discourse-calendar.js

```diff
@@ -514,50 +544,153 @@ function initializeDiscourseCalendar(api) {
+        : [detail];
+
+    groupedEventData.forEach((eventData) => {
+      let htmlContent = "";
+      let users = [];
+      let localEventNames = [];
+
+      Object.keys(eventData.localEvents)
+        .sort()
+        .forEach((key) => {
+          const localEvent = eventData.localEvents[key];
+          htmlContent += `<b>${key}</b>: ${localEvent.users
+            .map((u) => u.username)
+            .sort()
+            .join(", ")}<br>`;
+          users = users.concat(localEvent.users);
+          localEventNames.push(key);
+        });
 
-    const event = _buildEvent(detail);
-    event.classNames = ["grouped-event"];
+      const event = _buildEvent(eventData);
+      event.classNames = ["grouped-event"];
 
-    if (usernames.length > 2) {
-      event.title = `(${usernames.length}) ${localEventNames[0]}`;
-    } else if (usernames.length === 1) {
-      event.title = usernames[0];
-    } else {
-      event.title = isMobileView
-        ? `(${usernames.length}) ${localEventNames[0]}`
-        : `(${usernames.length}) ` + usernames.join(", ");
-    }
+      if (users.length > 2) {
+        event.title = `(${users.length}) ${localEventNames[0]}`;
+      } else if (users.length === 1) {
+        event.title = users[0].username;
+      } else {
+        event.title = isMobileView
+          ? `(${users.length}) ${localEventNames[0]}`
+          : `(${users.length}) ` + users.map((u) => u.username).join(", ");
+      }
 
-    if (localEventNames.length > 1) {
-      event.extendedProps.htmlContent = htmlContent;
-    } else {
-      if (usernames.length > 1) {
+      if (localEventNames.length > 1) {
         event.extendedProps.htmlContent = htmlContent;
       } else {
-        event.extendedProps.htmlContent = localEventNames[0];
+        if (users.length > 1) {
+          event.extendedProps.htmlContent = htmlContent;
+        } else {
+          event.extendedProps.htmlContent = localEventNames[0];
+        }
       }
-    }
 
-    calendar.addEvent(event);
+      calendar.addEvent(event);
+    });
   }
 
-  function _setDynamicCalendarEvents(calendar, post, fullDay) {
+  function _splitGroupEventByTimezone(detail, calendarTz) {
+    const calendarUtcOffset = moment.tz(calendarTz).utcOffset();
+    let timezonesOffsets = [];
+    let splittedEvents = [];
+
+    Object.values(detail.localEvents).forEach((event) => {
+      event.users.forEach((user) => {
+        const userUtcOffset = moment.tz(user.timezone).utcOffset();
+        const timezoneOffset = (calendarUtcOffset - userUtcOffset) / 60;
+        user.timezoneOffset = timezoneOffset;
+        timezonesOffsets.push(timezoneOffset);
+      });
+    });
+
+    [...new Set(timezonesOffsets)].forEach((offset) => {
+      let filteredLocalEvents = {};
+      let eventTimezones = [];
+
+      Object.keys(detail.localEvents).forEach((key) => {
+        const threshold =
+          siteSettings.split_grouped_events_by_timezone_threshold;
```

Comment:
I tried locally to set this to various numbers, but I no longer see the events being offset in the UI. Is there a regression? For example, `testuser5` is set to a timezone in Africa, but their events don't offset by ~5 hours, regardless of the threshold value (I tried 0, 3, 5). 

<img width="819" alt="image" src="https://github.com/discourse/discourse-calendar/assets/368961/861b118b-eff4-4d22-ad5b-7bd46ace278b">

Reply:
Does the other non-grouped events work? Do you have ` fullDay="true"`?
I don't think the new setting has much to do with the offset

Reply:
Here, I set up testuser5 as a user in Africa, so, different timezone. The markdown for the events for that user is this: 

```
[date=2023-09-28 timezone="Africa/Accra"]
```

THis shows as a day before in the calendar for a user with the timezone in Toronto. 

```
[date-range from=2023-09-30T10:00:00 to=2023-09-30T13:00:00 timezone="Africa/Accra"]
```

This shows as a full-day event on the 30th, no offset, and time range isn't respected AFAICT.

Reply:
I tried reproducing your issue, but it's working just fine for me.

This is a user with the timezone profile set to "Africa/Accra". Although that does not really matter much, what matters is the timezone attribute in the date. I posted both of your dates.

Viewing the calendar with a user from "America/Toronto" I see this:

![image](https://github.com/discourse/discourse-calendar/assets/66427541/9c8ddb58-15bf-4bb2-a828-9ba247979f63)

My calendar markdown is `[calendar weekends=true tzPicker="true" fullDay="true"] [/calendar]`

And yes, the time range is not respected. That's why it That's why it needs `fullDay="true"`. Maybe in the future we can add a way to also adjust the event length according to time, but I left it out of the scope of this PR.

Reply:
Ahh, `fullDay=true` was the issue... with it, it's now fixed. Apologies, I missed that on your previous comment.